marshal/marshal.go

```diff
@@ -28,91 +30,371 @@ const (
 	HeartbeatInterval = 60 // Measured in seconds.
 )
 
+// Marshaler is the coordinator type. It is designed to be used once globally and
```

Comment:
This comment ("designed to be used once globally") contradicts the comment in the pull request:

> Marshaler -> created once per (client, group) and is in charge of handling the state for that level. This is the main thing you create and interact with as a user.